Cherry-picks and release notes for stable-2.10.1 (#6029)
linkerd/linkerd2: CHANGES.md

# Changes

## stable-2.10.1

This stable release adds support for Apple Silicon M1 chips in the CLI. This release
also includes support for SMI's TrafficSplit `v1alpha2`.

It also contains a number of proxy enhancements: handling `FailedPrecondition` errors
gracefully, improved logging infrastructure to reduce memory pressure in high-connection
environments, improved inbound proxy's TLS detection from non-meshed workloads and
fixed a cache issue in the outbound proxy with ingress mode.

> On the Control-plane side, It includes a lot of improvements to destination: support
> for Host IP lookups, ignoring pods in "Terminating State". It also updates the
> proxy-injector to add opaque ports annotation to pods if their namespace has it set.

Reviewer: Using a colon to list two items isn't really necessary:
```suggestion
On the control-plane side, there have been several improvements to the
destination service such as support for Host IP lookups and ignoring pods
"Terminating State". It also updates the proxy-injector to add opaque ports
annotation to pods if their namespace has it set.
```
